client/chain_client.go

```diff
@@ -44,33 +44,52 @@ func NewChainClient(ccc *ChainClientConfig, homepath string, input io.Reader, ou
 		Codec:          MakeCodec(ccc.Modules),
 		Logger:         log.NewTMLogger(log.NewSyncWriter(output)),
 	}
-	if err := cc.Init(); err != nil {
+	if err := cc.initKeybase(); err != nil {
 		return nil, err
 	}
-	return cc, nil
-}
-
-func (cc *ChainClient) Init() error {
-	// TODO: test key directory and return error if not created
-	keybase, err := keyring.New(cc.Config.ChainID, cc.Config.KeyringBackend, cc.Config.KeyDirectory, cc.Input, cc.KeyringOptions...)
-	if err != nil {
-		return err
-	}
-	// TODO: figure out how to deal with input or maybe just make all keyring backends test?
-
 	timeout, _ := time.ParseDuration(cc.Config.Timeout)
 	rpcClient, err := NewRPCClient(cc.Config.RPCAddr, timeout)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	lightprovider, err := prov.New(cc.Config.ChainID, cc.Config.RPCAddr)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	cc.RPCClient = rpcClient
 	cc.LightProvider = lightprovider
+
+	return cc, nil
+}
+
+func NewChainClientWithRPCClient(ccc *ChainClientConfig, cli rpcclient.Client, homepath string, input io.Reader, output io.Writer, kro ...keyring.Option) (*ChainClient, error) {
+	ccc.KeyDirectory = keysDir(homepath, ccc.ChainID)
+	cc := &ChainClient{
+		KeyringOptions: kro,
+		Config:         ccc,
+		Input:          input,
+		Output:         output,
+		Codec:          MakeCodec(ccc.Modules),
+		Logger:         log.NewTMLogger(log.NewSyncWriter(output)),
+	}
+	if err := cc.initKeybase(); err != nil {
+		return nil, err
+	}
+	cc.RPCClient = cli
+	// TODO @renaynay: this will be a problem not to set the light provider if we want a light block
+	return cc, nil
+}
+
+func (cc *ChainClient) initKeybase() error {
+	// TODO: test key directory and return error if not created
+	keybase, err := keyring.New(cc.Config.ChainID, cc.Config.KeyringBackend, cc.Config.KeyDirectory, cc.Input, cc.KeyringOptions...)
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out how to deal with input or maybe just make all keyring backends test?
```

Comment:
I think we should make all keyring backends test backends and eliminate the option from the struct